src/lib/mongoBackend/safeMongo.cpp

```diff
@@ -106,6 +106,34 @@ std::string getStringField(const BSONObj& b, const std::string& field, const std
   return "";
 }
 
+
+
+/* ****************************************************************************
+*
+* getNumberField -
+*/
+double getNumberField(const BSONObj& b, const std::string& field, const std::string& caller, int line)
+{
+  if (b.hasField(field) && b.getField(field).type() == NumberDouble)
+  {
+    return b.getField(field).Number();
+  }
+
+  // Detect error
+  if (!b.hasField(field))
+  {
+    LM_E(("Runtime Error (double field '%s' is missing in BSONObj <%s> from caller %s:%d)", field.c_str(), b.toString().c_str(), caller.c_str(), line));
+  }
+  else
+  {
+    LM_E(("Runtime Error (field '%s' was supposed to be an double but type=%d in BSONObj <%s> from caller %s:%d)",
+          field.c_str(), b.getField(field).type(), b.toString().c_str(), caller.c_str(), line));
+  }
+  return -1;
```

Comment:
-1 is a perfectly valid `double`.
How does the caller detect this error?
Seems to be this function should return 0 on OK and send the double as an output parameter ... ?